jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/NameEnricher.java

```diff
@@ -48,11 +48,14 @@ public NameEnricher(JKubeEnricherContext buildContext) {
 
   @Override
   public void create(PlatformMode platformMode, KubernetesListBuilder builder) {
-    final String defaultName = getConfig(Config.NAME, JKubeProjectUtil.createDefaultResourceName(getContext().getGav().getSanitizedArtifactId()));
+    final String configuredName = getConfig(Config.NAME);
+    final String defaultName = JKubeProjectUtil.createDefaultResourceName(getContext().getGav().getSanitizedArtifactId());
```

Comment:
Why is this change necessary, isn't the getConfig(key, defaultValue) taking care of this?

Reply:
I wanted NameEnricher's `Config.NAME` configuration option to take precedence over currently configured name in resource. Resource `.metadata.name` is overwritten if `Config.Name` is non blank regardless of it's value. However, `.metadata.name` is only overwritten in the default case if its value is `null`/empty.

Reply:
OK, the current structure makes it very confusing to read.
So the idea is that the precedence is:
- getConfig(Config.NAME) ---> create_withCustomNameAndAlreadyExistingName_shouldOverrideExistingName
- resource.getName() ----> *1
- JKubeProjectUtil.createDefaultResourceName(getContext().getGav().getSanitizedArtifactId()) ----> *2

Are there tests for 1 & 2?

Reply:
I think you're right. for 2 there seems to be `createWithDefaultsInKubernetes` test which verifies opinionated name use case.

For 1, I'm going to add a test case